### ISSUE TITLE
Cut-out temperature same as cut-in temperature when Stratified Tank indirectly heated by boiler

### ISSUE THREAD

**Issue by ryandmeyer1017** [NREL/EnergyPlus#7281]
Issue overview
--------------
I have a simple Central DHW system model. I have a WaterHeater:Stratified object indirectly heated by a boiler. I am using the IndirectHeatPrimarySetpoint Source Side Flow Control Mode. A schedule defines the tank set point, or 'cut-out' temperature. I have a 5 F dead-band prescribed to define the 'cut-in' temperature. 

The tank appears to correctly request source side heat when the source side outlet temperature drops below the 'cut-in' temperature. However, the tank appears to incorrectly end its request for source side heat when the source side outlet temperature rises above the 'cut-in' temperature instead of the 'cut-out' temperature. In other words, the tank never reaches the 'cut-out' temperature in the attached file. An output html file is included to view data from the .eso file. 

When the WaterHeater:Stratified object is used in a standalone configuration in lieu of indirectly heated by the boiler, operation appears to be correct. The internal heaters stop  when the 'cut-out' temperature is reached. A sample file for this configuration is not included with this post. 

### Details
Some additional details for this issue (if relevant):
 - Platform: MacOS High Sierra 10.13.6
 - Version of EnergyPlus: 9.1.0
 - Unmethours link or helpdesk ticket number: N/A

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [√] Defect file added (link below)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)
[in_test.zip](https://github.com/NREL/EnergyPlus/files/3118553/in_test.zip)

**Comment by yzhou601** [comment]
Hi, Ryan, I happened to be exploring the source side of water heater tanks recently. As far as I'm concerned, when outlet temperature is between deadband(cut-in) temperature and setpoint(cut-out) temperature, E+ judges whether requesting source heat by checking the ``tank saved mode``(if ``FloatMode``, it closed source side, if ``HeatMode``, it turned on), which is always float in this case. However, the heater on/off control is following the desired logic. That's the reason why the standalone model worked out to be different. As for the tank mode, it seems to be a parameter more useful for mixed tank, since I didn't find anywhere else in stratified tank calculation used tank.mode information. So I guess it's simply kept as default (float) all the time. Therefore, I think if the default mode is changed for stratified tank, it might work. But it would require small changes to source code. I didn't know if there's any way to model it with existing code since I tried some changes but didn't work. Hope this would help and welcome to comment for further discussion.

**Comment by ryandmeyer1017** [comment]
Thanks Yueyue. I do not see any reference to Tank.Mode for the stratified tank either. As an aside, I was able to get the expected operation when swapping the WaterHeater:Stratified with a Mixed object.

**Comment by yzhou601** [comment]
@ryandmeyer1017  You're welcome, the mode is an attribute of tank in source code, great that you've solved the problem by replacing the object. I am also investigating indirect tank, do you mind sharing me with your idf with mixed tank?

**Comment by ryandmeyer1017** [comment]
Of course. I've included the idf file here. For larger storage tanks it is still more appropriate that the stratified tank is available for the same operation. The Mixed tank offers a workaround for now., but I'd like to keep the Stratified object issue reported here open. Let me know if you find anything else. 
 
[in_mixed.idf.zip](https://github.com/NREL/EnergyPlus/files/3132059/in_mixed.idf.zip)
